### PR TITLE
fix: Trigger FugitiveChanged event after hunk operations

### PR DESCRIPTION
**Pull request by wookayin**
A feature for #507.

- [x] Have you read [CONTRIBUTING.md](https://github.com/lewis6991/gitsigns.nvim/blob/HEAD/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by wookayin** [comment]
It might be debatable whether gitsigns should trigger FugitiveChanged directly, or we may want to define and introduce a custom user-autocmd event (please let me know of your preference!). This approach (the former), despite a bit strong coupling with fugitive, might seem OK because the plugin is already aware of fugitive.

File: teal/gitsigns/manager.tl [hunk 1]
```diff
@@ -309,6 +309,8 @@ local update0 = function(bufnr: integer, bcache: CacheEntry)
   summary.head = git_obj.repo.abbrev_head
   Status:update(bufnr, summary)
 
+  vim.cmd [[ doautocmd <nomodeline> User FugitiveChanged ]]
```
Comment (lewis6991) [inline teal/gitsigns/manager.tl:312]: This is the wrong place to do this. Update does not change the index, it only reads from it and applies decorations to the buffer. The autocmd needs to be triggered in actions.tl for each action that can alter the index.

Comment (wookayin) [reply teal/gitsigns/manager.tl:312]: Got it. I will update accordingly.

**Comment by echasnovski** [comment]
@wookayin, are you still working on this? I found myself in a situation when custom event for hunk updates will be really handy.

@lewis6991, in your other plugin you [use `GitsignsHunkUpdate` event](https://github.com/lewis6991/satellite.nvim/blob/25d0c59edab4892363c3cec47fc5f34769e5a242/lua/satellite/handlers/gitsigns.lua#L14). I didn't find any mention of this event within this repository. Did you have in mind the result of this PR or your some other work?

**Comment by lewis6991** [comment]
Superseded by #641

**Pull request closed without merging**